analysis/spectra_chisq/_model_spectra.py

```diff
@@ -0,0 +1,475 @@
+
+    print('______________________________ fit parameters: ', fit_params)
+
+    return fit_params
+
+
+def fit_to_obs_spectra():
+    """Extinct model to fit to the observed spectra. Tabulate results.
+    Plots the last result in the loop currently. TODO move plotting elsewhere
+
+    Output table columns:
+        - obj_id
+        - time
+        - amplitude
+        - a_v
+
+    TODO run this via `run_analysis.py`
+    """
+
+    # Create table to save fit parameters to
+    table = Table(names=('obj_id', 'time', 'amplitude', 'a_v'), dtype=('S20', 'S20', 'f8', 'f8'))
+
+    # Create model for Milky Way dust
+    mw_dust = sncosmo.CCM89Dust()
+    # SN model with Milky Way dust
+    sn_model = sncosmo.Model(source='hsiao', effects=[mw_dust], effect_names=['mw'], effect_frames=['obs'])
+
+    for oid in dr3.get_available_ids()[8:9]: #FIXME remove [:#] after done testing
+        # Get spectroscopic data for specific SN
+        spec_data = dr1.get_data_for_id(oid)
+        # Get observation times
+        times = spec_data['time']
+        # Get object specific parameters: photometric data and redshift
+        _photo_data = dr3.get_data_for_id(oid)
+        _redshift = _photo_data.meta['z']
+
+        # Loop over unique observation times
+        for _time in set(times):
+            # Get spectrum taken at specific `_time`
+            spectrum = spec_data[spec_data['time'] == _time]
+            # Observer-frame (?) flux TODO
+            _flux = spectrum['flux'] # Units: ergs/s/cm^2/Angstrom (flux per wavelength)
+            # From the README.txt in 'CSP_spectra_Folatelli+2013.tgz' on https://csp.obs.carnegiescience.edu/data:
+            # "WAVELENGTH IS EXPRESSED IN THE REST FRAME OF THE SUPERNOVA IN ALL CASES."
+            rest_frame_wave = spectrum['wavelength'] # Units: Angstrom
+            # Get the observer-frame wavelength, because sncosmo `.flux()` returns observer frame flux...
+            obs_frame_wave = rest_frame_wave * (1 + _redshift)
+            # Fit the parameters: amplitude and Av, in SN model
+            _amp, _av = get_fit_parameters(obj_id=oid, time=_time, model=sn_model, wave=obs_frame_wave, flux=_flux,
+                                           redshift=_redshift)
+
+            # TODO / FIXME many fits return initial parameter guesses
+            print('SN: {}, redshift: {}, time: {}, amplitude: {}, Av: {}'.format(oid, _redshift, _time, _amp, _av))
+
+            table.add_row((oid, _time, _amp, _av))
+
+    #table.write('extinction_fit_parameters.fits', format='fits', overwrite=False)
+
+    # Get flux from the SN model using the best-fit parameters
+    mdl_flux = get_model_flux(wave=obs_frame_wave, obj_id=oid, time=_time, model=sn_model,
+                              amplitude=_amp, a_v=_av, redshift=_redshift)
+    plot_spectra(x1=obs_frame_wave, y1=_flux, x2=obs_frame_wave, y2=mdl_flux)
+
+
+
+def fit_band_to_obs_photo():
```

Comment:
When a placeholder function isn't necessary to execute other code, it's usually better to make a single Todo list at the top of the file. This is less cluttered.